src/Chunker.php

```diff
@@ -53,92 +40,118 @@ protected function schedule_chunk(array|Iterator $chunkData): void
+
+        // fetch the data
+        $data = $chunk->get_data();
+
+        // Convert array to Generator
+        $generator = (function () use ($data) {
+            foreach ($data as $key => $value) {
+                yield $key => $value;
             }
         })();
 
-        $this->process_chunk_data($formattedDataGenerator);
+        $this->process_chunk_data($generator);
+    }
 
-        fclose($file);
+    /**
+     * Handles the actual data processing. Should be implemented in the class lowest in hierarchy.
+     *
+     * @param Generator<mixed> $chunkData The generator containing chunk data to process
+     * @return void
+     */
+    abstract function process_chunk_data(Generator $chunkData): void;
 
-        // Remove chunk file after sync
-        $unlink_result = unlink($chunk_file_path);
-        if ( $unlink_result === false ) {
-            throw new Exception("File '$chunk_file_path' could not be deleted!");
-        }
+    /**
+     * Schedules the cleanup job if not already scheduled.
+     * Creates a daily cron job at midnight to clean up old chunk data.
+     *
+     * @return void
+     */
+    public function schedule_chunk_cleanup(): void
+    {
+        if ( as_has_scheduled_action( 'ASP/Chunks/Cleanup' ) ) {
+			return;
+		}
+
+        // schedule the cleanup midnight every day
+		as_schedule_cron_action(
+			time(),
+			'0 0 * * *', 'ASP/Chunks/Cleanup'
+		);
     }
 
     /**
-     * Handles the actual data processing. Should be implemented in the class lowest in hierarchy
+     * Cleans the chunk data table from data with following properties:
+     * - older than 2 days (start)
+     * - status must be finished
      *
-     * @param Generator $chunkData
      * @return void
      */
-    abstract function process_chunk_data(Generator $chunkData): void;
+    public function cleanup_chunk_data(): void
+    {
+        /**
+         * Filters the number of days to keep chunk data.
+         *
+         * @param int $interval The interval (e.g., 14).
+         */
+        $days_interval = apply_filters( 'asp/chunks/cleanup/days', 14 );
+        $days_interval = (int) $days_interval * DAY_IN_SECONDS;
+        $cleanup_timestamp = (int) time() - $days_interval;
 
+        /**
+         * Filters the status of chunks to clean up.
+         *
+         * @param string $status The status to filter by (default: 'all').
+         */
+        $status = apply_filters('asp/chunks/cleanup/status', ProcessStatus::ALL->value);
+
+        $query = '';
+        if ( 'all' === $status ) {
+            $query = $this->db()->prepare(
+                "DELETE FROM {$this->get_chunks_table_name()} WHERE start < %f",
+                $cleanup_timestamp
+            );
+        } else {
+            $query = $this->db()->prepare(
+                "DELETE FROM {$this->get_chunks_table_name()} WHERE status = %s AND start < %f",
+                $status,
+                $cleanup_timestamp
+            );
+        }
```

Comment:
use enum. Either pass enum to filter or pass string to filter and parse back to enum. 
```php
ProcessStatus::from($status);
```